### ISSUE TITLE
add support installing in ~/.config/bash_it instead of ~/.bash_it

### ISSUE THREAD

**Issue by edvinasme** [Bash-it/bash-it#519]
Optionally, we could use $XDG_CONFIG_HOME

Please let me know what you think regarding this.

Currently if I clone in in `~/.config/bash_it` and run `install.sh` I get errors during install:

```

$ ~/.config/bash_it/install.sh 
Backup file already exists. Make sure to backup your .bashrc before running this installation.
Would you like to overwrite the existing backup? This will delete your existing backup file (/home/edv/.bashrc.bak) [y/N] y
Your original .bashrc has been backed up to .bashrc.bak
cp: cannot stat ‘/home/edv/.bash_it/template/bash_profile.template.bash’: No such file or directory
Copied the template .bashrc into ~/.bashrc, edit this file to customize bash-it
Do you use Jekyll? (If you don't know what Jekyll is, answer 'n') [y/N] n
Would you like to enable all, some, or no aliases? Some of these may make bash slower to start up (especially completion). (all/some/none) all
mkdir: cannot create directory ‘/home/edv/.bash_it/aliases/enabled’: No such file or directory
/home/edv/.config/bash_it/install.sh: line 48: [: too many arguments
ln: failed to create symbolic link ‘/home/edv/.bash_it/aliases/enabled/*’: No such file or directory
Would you like to enable all, some, or no plugins? Some of these may make bash slower to start up (especially completion). (all/some/none) all
mkdir: cannot create directory ‘/home/edv/.bash_it/plugins/enabled’: No such file or directory
/home/edv/.config/bash_it/install.sh: line 48: [: too many arguments
ln: failed to create symbolic link ‘/home/edv/.bash_it/plugins/enabled/*’: No such file or directory
Would you like to enable all, some, or no completion? Some of these may make bash slower to start up (especially completion). (all/some/none) all
mkdir: cannot create directory ‘/home/edv/.bash_it/completion/enabled’: No such file or directory
/home/edv/.config/bash_it/install.sh: line 48: [: too many arguments
ln: failed to create symbolic link ‘/home/edv/.bash_it/completion/enabled/*’: No such file or directory
Installation finished successfully! Enjoy bash-it!

```

**Comment by nwinkler** [comment]
I've had a look at the install script (and the rest of the project). There are less than five places where the `.bash_it` folder name is used. Changing these to use a dynamic location should be fairly straightforward.

* https://github.com/Bash-it/bash-it/blob/master/install.sh#L2 - this could be changed to use the current directory name, i.e. the one the `install.sh` script is started from (discussion here: http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in)
* https://github.com/Bash-it/bash-it/blob/master/install.sh#L39 - this should simply use `$BASH_IT` instead of the hardcoded path...
* https://github.com/Bash-it/bash-it/blob/master/template%2Fbash_profile.template.bash#L4 - this one is more tricky. Since it's a template which is copied as part of the installation, it should probably be replaced as part of the installation. The template could use a placeholder, and instead of copying the file (`install.sh`), we could copy the template and replace the placeholder with the real location (sed/awk or similar).

Other than that, it's just documentation.

Let me know if you would like to create a PR for this - this would be a nice addition/improvement.

Thanks for reporting this!

**Comment by edvinasme** [comment]
I will try to create a PR during the coming days when I've got some free time.

**Comment by nwinkler** [comment]
:+1:

**Comment by edvinasme** [comment]
sorry I have been busy, if someone else has time feel free to submit a PR. Otherwise I can do it next month.

**Comment by aaroncaito** [comment]
ln -s ~/custom/bash_it ~/bash_it  ?

**Comment by justin8** [comment]
Wrote a PR for this:
https://github.com/Bash-it/bash-it/pull/589

**Comment by edvinasme** [comment]
> Wrote a PR for this:
> #589

@justin8 Thanks. I've made a small comment regarding your PR. Could someone from the maintainers comment if this can be merged?

> ln -s ~/custom/bash_it ~/bash_it ?

@aaroncaito Symlinking does not have anything to do with initial problem of installing bash-it in other directory than `~/.bash_it`.

**Comment by justin8** [comment]
The symlinking does not; could be in a seperate PR. but if you run install twice in a row (which I did many many more times than twice making it all work ;)) it has errors on the console. Alternative might be to suppress the messages, but they're ugly; and it's linking for things inside of the repo/install directory, so it should be fine to overwrite state in there.

**Comment by nwinkler** [comment]
I've started to test the PR - give me a moment to try this in various scenarios.

**Comment by nwinkler** [comment]
Found a stray `exit` statement in the install script - see my comment in #589

**Comment by nwinkler** [comment]
Implemented in #589
